needls/needs_store.py

```diff
@@ -35,19 +36,29 @@ def is_setup(self) -> bool:
 
     def set_docs_root(self, docs_root: str) -> None:
         if not os.path.exists(docs_root):
-            raise ValueError(f"Docs root directory not found: {docs_root}")
+            raise FileNotFoundError(f"Docs root directory not found: {docs_root}")
         self.docs_root = docs_root
 
     def set_declared_types(self) -> None:
         module_name = "conf"
         work_dir = os.getcwd()
         os.chdir(self.docs_root)
+
+        logging.info("Loading need_types from conf.py...")
+
         spec = importlib.util.spec_from_file_location(
             module_name, os.path.join(self.docs_root, "conf.py")
         )
-        module = importlib.util.module_from_spec(spec)
-        sys.modules[module_name] = module
-        spec.loader.exec_module(module)
+        if spec is not None:
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[module_name] = module
+            try:
+                spec.loader.exec_module(module)
+            except Exception as e:
+                logging.error(f"Failed to exccute module {module} -> {e}")
+        else:
```

Comment:
Simplify this:

Check and maybe raise exception, then the rest of the code does not need to be in an if clause:

```
if spec is None:
  raise Import Error(...)
  
module = importlib....
....
```